browser/tor/tor_profile_service.cc

```diff
@@ -56,15 +56,63 @@ void TorProfileService::RegisterLocalStatePrefs(PrefRegistrySimple* registry) {
   const std::string tor_proxy_uri =
       std::string(kTorProxyScheme) + std::string(kTorProxyAddress) + ":" + port;
   registry->RegisterStringPref(prefs::kTorProxyString, tor_proxy_uri);
-  // Tor for group policy. Only can be true by group policy.
+  // kTorDisabled is managed prefs.
+  // User can only change it via settiongs option if it's not managed.
   registry->RegisterBooleanPref(prefs::kTorDisabled, false);
+  // kTorDisabledAtNextLaunching has the value that user changed via settings.
+  // If this value is changed, it is exposed to settings page. At the next
+  // launching, this value is copied to kTorDisabled.
+  registry->RegisterBooleanPref(prefs::kTorDisabledAtNextLaunching, false);
```

Comment:
rather than adding another pref for this, do we have or can we add a check to see if Tor is currently running? Then you can compare the running state to the pref state? I didn't understand what you were trying to do before with the static vars, but now that I do understand I think maybe there's still a better way to handle it.

Reply:
I wanted to do one more thing with this new pref.
I think  IsTorDisabled() should not change its return value during the runtime.
Otherwise, if user disables tor, browser could think tor is disabled now but tor will be disabled(and deleted its component) after relaunching. So, I needed to store user's choice persistently instead of changing `kTorDisabled` prefs. and then this cached value is copied to `kTorDisabled`. I think this is only achieved by using another persistent value.

Reply:
Ah.. I think this could be achieved w/o another pref.
Whether tor is running or not could checked tor component is registered or not.
At the startup, Tor component is registered by checking `kTorDisabled`.
After that, whether tor is running or not could checked tor component is registered or not.
And `kTorDisabled` is used to store user's change.
I think this will work :)

Reply:
`kTorDisabledAtNextLaunching` is deleted.